### PR TITLE
re-enable counter_5clock,sdc_controller, lut_adder tests

### PR DESCRIPTION
**Pull request by coolbreeze413**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

Re-Enable remaining advanced quiicklogic regression tests: counter_5clock,sdc_controller, lut_adder tests

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] VPR
> - [ ] Tileable routing architecture generator
> - [ ] OpenFPGA libraries
> - [ ] FPGA-Verilog
> - [ ] FPGA-Bitstream
> - [ ] FPGA-SDC
> - [ ] FPGA-SPICE
> - [ ] Flow scripts
> - [ ] Architecture library
> - [ ] Cell library
> - [ ] Documentation
> - [x] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

**Comment by coolbreeze413** [comment]
@tangxifan tests pass locally, however strangely CI complains about share/ directory:

```
ERROR (00_or2_MIN_ROUTE_CHAN_WIDTH) - command /__w/OpenFPGA/OpenFPGA/yosys/install/bin/yosys yosys.ys
ERROR (00_or2_MIN_ROUTE_CHAN_WIDTH) - -->>ERROR: init_share_dirname: unable to determine share/ directory!

```
Am not sure yet why this happens, please check if I am missing something.

**Comment by coolbreeze413** [comment]
Locally, (running on Ubuntu 20.04), I get the share dir path correctly: `/work/openfpga/openfpga-gh/yosys/install/bin/../share/yosys/`
which appears from here: https://github.com/YosysHQ/yosys/blob/dca8fb54aa625f1600e2ccb16f9763c6abfa798f/kernel/yosys.cc#L867
Perhaps the docker images need to be updated in any way?
@ganeshgore Can you please also take a look and advice?

**Comment by tangxifan** [comment]
@ganeshgore Still have some issues here.

**Comment by ganeshgore** [comment]
@tangxifan  Wait for master branch to finish CI test, that will update docker image and then re-run this

**Comment by tangxifan** [comment]
@ganeshgore No worries. Retrying it now.

**Comment by tangxifan** [comment]
@ganeshgore Error remains. Can you double check the docker image? See if we do have the files in?

**Comment by coolbreeze413** [comment]
@tangxifan @ganeshgore I checked the latest master docker image, and I do see that the yosys/share/ directory is still empty.

I checked the upload-artifact step from the gcc-8 compilation which I see is the set of files that should go into the docker images, and the latest run here: https://github.com/lnis-uofu/OpenFPGA/actions/runs/1528939558 
has the artifact zip file containing the yosys/share/yosys directory as expected.
The latest artifact zip is here: https://github.com/lnis-uofu/OpenFPGA/suites/4527890086/artifacts/121221714
So, I think that the fix is working at this line : https://github.com/lnis-uofu/OpenFPGA/blob/master/.github/workflows/build.yml#L173

We need to see why the files from the uploaded artifact are not in the docker image.
Will investigate this some more from my side as well.

**Comment by ganeshgore** [comment]
@coolbreeze413 Yes you are right. I also observed that the artifact has all the expected files but it's not getting copied to the docker image. I suppose we need to debug this line/step.  
https://github.com/lnis-uofu/OpenFPGA/blob/9f1d47b6c0b835781b453650027284419dc08a8c/.github/workflows/build.yml#L184
I am running one CI run to see what goes into this step

**Comment by coolbreeze413** [comment]
@ganeshgore ok. If the upload-artifact contains the files, the download-artifact should also contain that, as they both internally use the same underlying package. It may be that the docker image build step is missing something or not taking files from the location we expect.
Please let me know if you find anything in the CI run you trigger, I can help debug as well.

**Comment by ganeshgore** [comment]
I guess the docker COPY option is not copying files to the docker container 
I added this option to the docker build 
https://github.com/lnis-uofu/OpenFPGA/blob/8bb4e89b54f2fdc6bdac58729d61a8deb1c8f58c/docker/Dockerfile.master#L5
it is missing inside the container  
https://github.com/lnis-uofu/OpenFPGA/runs/4391367472?check_suite_focus=true#step:8:232

However, I could see the artifacts are downloaded correctly with this option added 
https://github.com/lnis-uofu/OpenFPGA/blob/8bb4e89b54f2fdc6bdac58729d61a8deb1c8f58c/.github/workflows/build.yml#L192

https://github.com/lnis-uofu/OpenFPGA/runs/4391367472?check_suite_focus=true#step:4:7899

In short, the COPY command in `dockerfile.master` is  missing something

**Comment by ganeshgore** [comment]
@coolbreeze413 Solved, following line, should not have `**` sign at the end 
https://github.com/lnis-uofu/OpenFPGA/blob/8bb4e89b54f2fdc6bdac58729d61a8deb1c8f58c/.dockerignore#L15
Let me make separate PR and update the docker image, then you can rerun CI for this PR

**Comment by coolbreeze413** [comment]
@ganeshgore nice! Thanks! I will rerun the CI after the docker images are regenerated.

**Comment by coolbreeze413** [comment]
Re-running CI after the docker image fixes - tests are now passing.

**Review by tangxifan — APPROVED** [review]
